tests/system/test_signal_crashes.py

```diff
@@ -296,9 +305,31 @@ def create_test_process(self, command=None, uid=None, args=None):
             else:
                 break
 
-        time.sleep(0.3)  # needs some more setup time
+        # sleep command needs extra time to get into "sleeping" state
+        if command == self.TEST_EXECUTABLE:
+            wait_for_sleep_cmd(process.pid)
+
         return process
 
+    def test_create_test_sleep_process(self) -> None:
+        """Test create_test_process() helper method with sleep process."""
+        proc: subprocess.Popen = self.create_test_process()
+        try:
+            self.assertEqual(psutil.Process(proc.pid).status(), "sleeping")
+        finally:
+            proc.kill()
+            proc.wait()
+
+    @patch("tests.helper.wait_for_sleep_cmd")
+    def test_create_test_non_sleep_process(self, wait_sleep_mock: MagicMock) -> None:
+        """Test create_test_process() helper method with non-sleep process."""
+        with patch("os.access"):
+            with patch("subprocess.Popen"):
+                with patch("tests.system.test_signal_crashes.open"):
```

Comment:
You can put those in one `with` statement:
```suggestion
        with (
            patch("os.access"),
            patch("subprocess.Popen"),
            patch("tests.system.test_signal_crashes.open")
        ):
```
Or you can put those as patch decorators:
```python
            @patch("os.access", MagicMock())
            @patch("subprocess.Popen", MagicMock())
            @patch("tests.system.test_signal_crashes.open", MagicMock())
```

Reply:
Using patch decorators will cause the linter to complain about unused arguments, which is why I went with the context manager approach.

I had considered the single `with` statement, but I was concerned about how far back the CI would run since that's only supported in python 3.10. Although it looks like we only need to support jammy, which uses python3.10. So I'll go ahead and update it to use that.

Reply:
Regarding linters: There are two solutions:
1. Specify the mock object so that the test function is not called with the mock object as parameter: `@patch("tests.system.test_signal_crashes.open", MagicMock())`
2. Check that these mocks were called.